health-monitors/gpu-health-monitor/gpu_health_monitor/tests/test_dcgm_watcher/test_dcgm.py

```
@@ -1234,3 +1245,279 @@ def test_other_nvlink_error_code_not_suppressed(self, tmp_path: Path) -> None:
+
+        for _ in range(2):
+            with watchdog.probe("dcgm_health_check"):
+                time.sleep(0.05)
+                watchdog.poll_once()
+
+        assert len(hangs) == 2
+
+    def test_run_returns_when_exit_is_set(self):
+        hangs, on_hang = self._collector()
+        watchdog = dcgm.ProbeWatchdog(10.0, on_hang)
+        exit = Event()
```

Ruff (0.16.1)

[error] Variable `exit` is shadowing a Python builtin

(A001)